segmentation/include/pcl/segmentation/impl/extract_labeled_clusters.hpp

```diff
@@ -101,26 +153,54 @@ pcl::extractLabeledEuclideanClusters (const PointCloud<PointT> &cloud,
 
     // If this queue is satisfactory, add to the clusters
     if (seed_queue.size () >= min_pts_per_cluster && seed_queue.size () <= max_pts_per_cluster)
-    {
-      pcl::PointIndices r;
-      r.indices.resize (seed_queue.size ());
-      for (std::size_t j = 0; j < seed_queue.size (); ++j)
-        r.indices[j] = seed_queue[j];
-
-      std::sort (r.indices.begin (), r.indices.end ());
-      r.indices.erase (std::unique (r.indices.begin (), r.indices.end ()), r.indices.end ());
+    { 
+      if ( current_label_cluster_it == labeled_clusters.end() )  // label not found in map, add
+      {
+        std::vector<pcl::Indices> v = {};
+        v.emplace_back(std::move(seed_queue));
+        labeled_clusters.emplace(current_label, std::move(v));
+      }
+      else  // label already exists in map; append seed_queue to label vector
+        current_label_cluster_it->second.emplace_back(std::move(seed_queue));
```

Comment:
```suggestion
      labeled_clusters[current_label].emplace_back(std::move(seed_queue));
```

Reply:
Not sure how I missed that one -- I guess it's been a while since I used `map`.  Thanks!